### PR TITLE
added check for links count

### LINKED ISSUE #7011: LinkFree TotalLinkClick get increase on own's click
### Description

The Total Link Click in Profile get increase on one's own click.

If it is supposed  to be that way then i get.

### Screenshots

Screenshot from 2023-05-19 21-37-20

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by Akhilbisht798**
## Fixes Issue

closes #7011 

## Check List (Check all the applicable boxes) 

- [x] My code follows the code style of this project.
- [ ] My change requires changes to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] All new and existing tests passed.
- [x] This PR does not contain plagiarized content.
- [x] The title of my pull request is a short description of the requested changes.

## Note to reviewers

<a href="https://gitpod.io/#https://github.com/EddieHubCommunity/LinkFree/pull/7098"></a>

### CODE REVIEW

**Review by eddiejaoude — CHANGES_REQUESTED** [review]
Thank you for your contribution 👍 

I left an inline comment, I think the code can be simplified and reduced

File: pages/api/users/[username]/links/[url].js [hunk 1]
```diff
@@ -6,6 +9,12 @@ export default async function handler(req, res) {
   await connectMongo();
 
   const { username, url } = req.query;
+  let isOwner = false;
+  const session = await getServerSession(req, res, authOptions);
+
+  if (session && session.username === username) {
+    isOwner = true;
```
[the remark this answers is not in the document]
Comment (Akhilbisht798) [reply pages/api/users/[username]/links/[url].js:16]: Oh. thats a great idea. We can remove all the checks for below by simply returning early.

**Comment by Akhilbisht798** [comment]
Hey, I have made the changes you have suggested in the file.

**Review by eddiejaoude — APPROVED** [review]
Thank you 👍

**Pull request merged**